apps/federation/lib/Controller/SettingsController.php

```diff
@@ -45,33 +57,69 @@ public function addServer(string $url): DataResponse {
 	}
 
 	/**
-	 * Add server to the list of trusted Nextclouds.
+	 * Add server to the list of trusted Nextcloud servers
+	 *
+	 * @param int $id The ID of the trusted server to remove
+	 * @return DataResponse<Http::STATUS_OK, array{id: int}, array{}>|DataResponse<Http::STATUS_NOT_FOUND, array{message: string}, array{}>
+	 *
+	 * 200: Server removed successfully
+	 * 404: Server not found at the given ID
 	 */
 	#[AuthorizedAdminSetting(settings: Admin::class)]
+	#[ApiRoute(verb: 'DELETE', url: '/trusted-servers/{id}', requirements: ['id' => '\d+'])]
 	public function removeServer(int $id): DataResponse {
-		$this->trustedServers->removeServer($id);
-		return new DataResponse();
+		try {
+			$this->trustedServers->getServer($id);
+		} catch (\Exception $e) {
+			throw new OCSNotFoundException($this->l->t('No server found with ID: %s', [$id]));
+		}
+
+		try {
+			$this->trustedServers->removeServer($id);
+			return new DataResponse(['id' => $id]);
+		} catch (\Exception $e) {
+			$this->logger->error($e->getMessage(), ['e' => $e]);
+			throw new OCSException($this->l->t('Could not remove server'), Http::STATUS_INTERNAL_SERVER_ERROR);
+		}
 	}
 
 	/**
-	 * Check if the server should be added to the list of trusted servers or not.
+	 * List all trusted servers
 	 *
-	 * @throws HintException
+	 * @return DataResponse<Http::STATUS_OK, list<array{id: int, status: int, url: string}>, array{}>
+	 *
+	 * 200: List of trusted servers
+	 */
+	#[AuthorizedAdminSetting(settings: Admin::class)]
+	#[ApiRoute(verb: 'GET', url: '/trusted-servers')]
+	public function getServers(): DataResponse {
+		$servers = $this->trustedServers->getServers();
+
+		// obfuscate the shared secret
+		$servers = array_map(function ($server) {
```

Comment:
nit: introduce a new variable instead of repurposing the old one.

```suggestion
		$redactedServers = array_map(function ($server) {
```